boot/freeldr/freeldr/arch/uefi/uefihw.c

```diff
@@ -0,0 +1,151 @@
+            sizeof(ACPI_BIOS_DATA) - sizeof(BIOS_MEMORY_MAP);
+
+        /* Set 'Configuration Data' value */
+        PartialResourceList = FrLdrHeapAlloc(sizeof(CM_PARTIAL_RESOURCE_LIST) +
+                                             TableSize, TAG_HW_RESOURCE_LIST);
+        if (PartialResourceList == NULL)
+        {
+            ERR("Failed to allocate resource descriptor\n");
+            return;
+        }
+
+        RtlZeroMemory(PartialResourceList, sizeof(CM_PARTIAL_RESOURCE_LIST) + TableSize);
+        PartialResourceList->Version = 0;
+        PartialResourceList->Revision = 0;
+        PartialResourceList->Count = 1;
+
+        PartialDescriptor = &PartialResourceList->PartialDescriptors[0];
+        PartialDescriptor->Type = CmResourceTypeDeviceSpecific;
+        PartialDescriptor->ShareDisposition = CmResourceShareUndetermined;
+        PartialDescriptor->u.DeviceSpecificData.DataSize = TableSize;
+
+        /* Fill the table */
+        AcpiBiosData = (PACPI_BIOS_DATA)&PartialResourceList->PartialDescriptors[1];
+
+        if (Rsdp->revision > 0)
+        {
+            TRACE("ACPI >1.0, using XSDT address\n");
+            AcpiBiosData->RSDTAddress.QuadPart = Rsdp->xsdt_physical_address;
+        }
+        else
+        {
+            TRACE("ACPI 1.0, using RSDT address\n");
+            AcpiBiosData->RSDTAddress.LowPart = Rsdp->rsdt_physical_address;
+        }
+
+        AcpiBiosData->Count = FreeldrDescCount;
+        memcpy(AcpiBiosData->MemoryMap, EfiMemoryMap,
+            FreeldrDescCount * sizeof(BIOS_MEMORY_MAP));
+
+        TRACE("RSDT %p, data size %x\n", Rsdp->rsdt_physical_address,
+            TableSize);
+
+        /* Create new bus key */
+        FldrCreateComponentKey(SystemKey,
+                               AdapterClass,
+                               MultiFunctionAdapter,
+                               0x0,
+                               0x0,
+                               0xFFFFFFFF,
+                               "ACPI BIOS",
+                               PartialResourceList,
+                               sizeof(CM_PARTIAL_RESOURCE_LIST) + TableSize,
+                               &BiosKey);
+
+        /* Increment bus number */
+        (*BusNumber)++;
+    }
+}
+
+PCONFIGURATION_COMPONENT_DATA
+UefiHwDetect(VOID)
+{
+    PCONFIGURATION_COMPONENT_DATA SystemKey;
+    ULONG BusNumber = 0;
+
+    TRACE("DetectHardware()\n");
+
+    /* Create the 'System' key */
+    FldrCreateSystemKey(&SystemKey);
+#if defined(_M_IX86) || defined(_M_AMD64) /* Taken from Windows 11 */
+    FldrSetIdentifier(SystemKey, "AT/AT COMPATIBLE");
+#elif defined(_M_IA64) /* Taken from Windows XP 64-bit*/
+    FldrSetIdentifier(SystemKey, "Intel Itanium processor family");
+#elif defined(_M_ARM) || defined(_M_ARM64) /* Taken from Windows RT/11 */
+    FldrSetIdentifier(SystemKey, "ARM processor family");
+#else
+    #error "Please define a system key for your architecture"
```

Comment:
Quotes are not required.
```suggestion
    #error Please define a system key for your architecture
```